### PR TITLE
Add Network Profile swagger and examples

### PR DESCRIPTION
**Pull request by Flip-Liquid**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jcgknudson** [comment]
@jianghaolu , would you mind reviewing as well? Unable to request review since I don't have write access on the repo :)

**Comment by lmazuel** [comment]
@ravbhatnagar from ARM as well for review

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+          "type": "string",
+          "description": "ARM Uri for the container."
+        }
+      },
+      "description": "Container resource properties."
+    },
+    "Container": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ContainerPropertiesFormat",
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json]: You probably want to use x-ms-client-flatten to make better SDK.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2016-12-01/applicationGateway.json#L511

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The reference of the subnet resource to create a contatainer network interface ip configruation."
+        }
+      },
+      "description": "IP configruation profile properties."
+    },
+    "IPConfigurationProfile": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/IPConfigurationProfilePropertiesFormat",
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json]: You probably want to use x-ms-client-flatten to make better SDK.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2016-12-01/applicationGateway.json#L511

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+          },
+          "description": "A list of container network interfaces created from this container network interface configuration."
+        }
+      },
+      "description": "Container network interface configuration properties."
+    },
+    "ContainerNetworkInterfaceConfiguration": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ContainerNetworkInterfaceConfigurationPropertiesFormat",
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json]: You probably want to use x-ms-client-flatten to make better SDK.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2016-12-01/applicationGateway.json#L511

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+        "container": {
+          "$ref": "#/definitions/Container",
+          "description": "Reference to the conatinaer to which this container network interface is attached."
+        }
+      }
+    },
+    "ContainerNetworkInterface": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ContainerNetworkInterfacePropertiesFormat",
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json]: You probably want to use x-ms-client-flatten to make better SDK.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2016-12-01/applicationGateway.json#L511

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+          },
+          "description": "List of chid container network interface configurations."
+        }
+      },
+      "description": "Network profile properties."
+    },
+    "NetworkProfile": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/NetworkProfilePropertiesFormat",
```
Comment (lmazuel) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json]: You probably want to use x-ms-client-flatten to make better SDK.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2016-12-01/applicationGateway.json#L511

**Review by lmazuel — APPROVED** [review]
Sounds good. A few remark about flatenning, this would avoid `myvariable.properties.myname` to be simply `myvariable.myname`

**Review by tjprescott — CHANGES_REQUESTED** [review]
Please implement @lmazuel's flattening suggestions. Without them, then Python SDK is a mess.

**Review by tjprescott — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
This was reviewed in person. Signing off!

**Comment by lmazuel** [comment]
Hi @jcgknudson 
Please also fix the model validation errors:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/424973532

This could mean that the examples are wrong, or the Swagger is wrong. Be sure that the example reflects the exact truth of the service. Once it's done, if there is still issues, this means your Swagger is wrong (because if example tells the truth, then it should match).
Let me know if something is unclear.
Thank you!

**Comment by jcgknudson** [comment]
Hey @lmazuel, fixed the issues with the model validation, sorry about that.
There were a few fields omitted in the model, nothing too big.

**Pull request merged**